icu4j/tools/build/src/main/java/com/ibm/icu/dev/tool/errorprone/ParseMavenOutForErrorProne.java

```diff
@@ -0,0 +1,109 @@
+        Map<String, List<ErrorEntry>> errorReport = new TreeMap<>();
+        ErrorEntry currentError = null;
+
+        if (baseDir != null) {
+            baseDir = baseDir.replace('\\', '/');
+        }
+        int currentLine = 0;
+        for (String line : Files.readAllLines(Paths.get(fileName), StandardCharsets.UTF_8)) {
+            currentLine++;
+            Matcher m = pattern.matcher(line);
+            if (m.find()) {
+                String path = line.substring(m.start(1), m.end(1)).replace('\\', '/');
+                if (baseDir != null) {
+                    if (path.startsWith(baseDir)) {
+                        path = path.substring(baseDir.length());
+                    }
+                }
+                // If we already had an error report in progress, save it
+                addErrorToReport(errorReport, currentError);
+                currentError =
+                        new ErrorEntry(
+                                path,
+                                Integer.parseInt(line.substring(m.start(2), m.end(2))), // line
+                                Integer.parseInt(line.substring(m.start(3), m.end(3))), // column
+                                line.substring(m.start(4), m.end(4)), // error code
+                                line.substring(m.start(5), m.end(5)) // message
+                        );
+            } else if (line.startsWith("  Did you mean ")) {
+                if (currentError == null) {
+                    error(fileName, currentLine, line, "Parse error: unexpected 'Did you mean' ");
+                } else {
+                    currentError.addExtra(line.trim());
+                }
+            } else if (line.startsWith("    (see https://") && line.endsWith(")")) {
+                if (currentError == null) {
+                    error(fileName, currentLine, line, "Parse error: unexpected '(see <url>)'");
+                } else {
+                    // 9 is the length of "    (see "
+                    currentError.url = line.substring(9, line.length() - 1);
+                }
+            } else if (line.equals(
+                    "[WARNING] Unable to autodetect 'javac' path, using 'javac' from the"
+                        + " environment.")) {
+                currentError = addErrorToReport(errorReport, currentError);
+            } else if (line.startsWith("[INFO]")) {
+                currentError = addErrorToReport(errorReport, currentError);
+            } else {
+                error(fileName, currentLine, line, "Parse error: I don't know what this is");
+                currentError = addErrorToReport(errorReport, currentError);
+            }
+        }
+        // In case we had an
+        currentError = addErrorToReport(errorReport, currentError);
+        return errorReport;
+    }
+
+    static ErrorEntry addErrorToReport(Map<String, List<ErrorEntry>> errorReport,
```

Comment:
seems clearer to return void here and just set currentError = null after the second call site

Reply:
Renamed the method and added a comment explaining why we return `null`